rpc/handler.go

```diff
@@ -479,33 +513,7 @@ func (h *handler) runMethod(ctx context.Context, msg *jsonrpcMessage, callb *cal
 	stream.WriteObjectField("result")
 	_, err := callb.call(ctx, msg.Method, args, stream)
 	if err != nil {
-		//return msg.errorResponse(err)
-
-		stream.WriteMore()
-		stream.WriteObjectField("error")
-		stream.WriteObjectStart()
-		stream.WriteObjectField("code")
-		ec, ok := err.(Error)
-		if ok {
-			stream.WriteInt(ec.ErrorCode())
-		} else {
-			stream.WriteInt(defaultErrorCode)
-		}
-		stream.WriteMore()
-		stream.WriteObjectField("message")
-		stream.WriteString(fmt.Sprintf("%v", err))
-		de, ok := err.(DataError)
-		if ok {
-			stream.WriteMore()
-			stream.WriteObjectField("data")
-			data, derr := json.Marshal(de.ErrorData())
-			if derr == nil {
-				stream.Write(data)
-			} else {
-				stream.WriteString(fmt.Sprintf("%v", derr))
-			}
-		}
-		stream.WriteObjectEnd()
+		HandleError(err, stream)
```

Comment:
Error here was written as top-level object field, not as sub-object.

Reply:
:hankey: Ok, I've addressed this in 4c640cf5cdb5d0814c72c42bfa0769f438d7df96. It should now set result to null and set the error object in the top-level JSON. Which seems to make the most sense in these cases.